### PR TITLE
[MRG] speedup get_rejection_thresholds

### PR DESCRIPTION
**Pull request by jasmainak**
closes https://github.com/autoreject/autoreject/issues/75.

### CODE REVIEW

File: autoreject/autoreject.py [hunk 1]
```diff
@@ -158,8 +158,9 @@ def get_rejection_threshold(epochs):
     rejection dictionary.
     """
     reject = dict()
+    epochs = epochs.copy()
+    epochs.decimate(decim=decim)
```
Comment (agramfort) [inline autoreject/autoreject.py]: I will call this only if decim=1

Comment (agramfort) [reply autoreject/autoreject.py]: and maybe you need to copy only if decim > 1

File: autoreject/autoreject.py [hunk 2]
```diff
@@ -191,12 +185,39 @@ def get_rejection_threshold(epochs):
...
+        X = this_epoch.get_data()
+        deltas = np.array([np.ptp(d, axis=1) for d in X])
+        all_threshes = np.sort(deltas.max(axis=1))
+        n_epochs, n_channels, n_times = X.shape
+
+        print('Estimating rejection dictionary for %s' % ch_type)
+        cache = dict()
+        est = GlobalAutoReject()
+        est.n_channels = n_channels
+        est.n_times = n_times
```
Comment (agramfort) [inline autoreject/autoreject.py]: est = GlobalAutoReject(n_channels=n_channels, n_times=n_times)

should work

Comment (jasmainak) [reply autoreject/autoreject.py]: True ... it's cruft of copy paste. I'll fix

File: autoreject/autoreject.py [hunk 3]
```diff
@@ -158,8 +158,10 @@ def get_rejection_threshold(epochs):
     rejection dictionary.
     """
     reject = dict()
+    if decim > 1:
```
Comment (agramfort) [inline autoreject/autoreject.py]: decim param is not tested

**Comment by jasmainak** [comment]
Merging so we can move on with our paper. Thanks @agramfort for the review

**Pull request merged**